src/object/flattenObject.ts

```diff
@@ -46,7 +46,12 @@ function flattenObjectImpl(object: object, prefix = ''): Record<string, any> {
 
     if (Array.isArray(value)) {
       for (let index = 0; index < value.length; index++) {
-        result[`${prefixedKey}.${index}`] = value[index];
+        const valueKey = `${prefixedKey}.${index}`;
+        if (typeof value[index] === 'object' && value[index] !== null) {
```

Comment:
I think non-object filter needed on top of flattenObjectImpl function, but not sure to code convention of es-toolkit.

Reply:
Actually I think we have a simpler logic here; I made a little modification.